libs/wingsdk/src/shared-aws/api.ts

```diff
@@ -0,0 +1,64 @@
+import { IResource } from "../std";
+
+/**
+ * A shared interface for AWS api.
+ */
+export interface IAwsApi {
+  /**
+   * RestApi arn
+   */
+  arn(): string;
+
+  /**
+   * RestApi id
+   */
+  restApiId(): string;
+
+  /**
+   * RestApi id
+   */
+  restApiName(): string;
+
+  /**
+   * RestApi stage name
+   */
+  stageName(): string;
+
+  /**
+   * Invoke URL
+   */
+  invokeUrl(): string;
+
+  /**
+   * RestApi deployment id
+   */
+  deploymentId(): string;
+}
+
+/**
+ * A helper class for working with AWS apis.
+ */
+export class Api {
+  /**
+   * If the api is an AWS RestApi, return a helper interface for
+   * working with it.
+   * @param api The cloud.Api.
+   */
+  public static from(api: IResource): IAwsApi | undefined {
```

Comment:
I think this should be more strongly typed:

```suggestion
  public static from(api: cloud.Api): IAwsApi | undefined {
```

Reply:
Basically, I used the `IResource` because I use the `IAwsBucket` for the `Bucket` and the `Website`, and took advantage of `Bucket.from` to check if it is an AWS resource.

However, I can create a specific one for the Website

Reply:
Let's do it. The stronger the type the better.